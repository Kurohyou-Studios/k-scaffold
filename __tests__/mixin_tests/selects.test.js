import * as fs from 'fs/promises';

import { describe,it,expect } from 'vitest';

import { k } from '../testFramework';

const htmlURL = new URL('../../test_sheet/build/selects.test.html',import.meta.url);

describe('selects entry mixins',()=>{
  it('Should create the selects mixins',async ()=>{
    const html = await fs.readFile(htmlURL,'utf8');
    expect(html).toMatchSnapshot();
  })
})