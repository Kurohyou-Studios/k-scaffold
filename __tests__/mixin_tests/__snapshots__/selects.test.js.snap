// Vitest Snapshot v1

exports[`selects entry mixins > Should create the selects mixins 1`] = `
"
<select name=\\"attr_select\\" title=\\"@{select}\\">
  <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
  <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
</select>
<label class=\\"input-label\\"><span class=\\"input-label__text\\" data-i18n=\\"select label\\"></span>
  <select class=\\"input-label__input\\" name=\\"attr_labelled_select\\" title=\\"@{labelled_select}\\">
    <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
    <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
  </select>
</label>
<fieldset class=\\"repeating_basic-fieldset\\">
  <select name=\\"attr_select\\" title=\\"@{repeating_basic-fieldset_$X_select}\\">
    <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
    <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
  </select>
  <label class=\\"input-label\\"><span class=\\"input-label__text\\" data-i18n=\\"select label\\"></span>
    <select class=\\"input-label__input\\" name=\\"attr_labelled_select\\" title=\\"@{repeating_basic-fieldset_$X_labelled_select}\\">
      <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
      <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
    </select>
  </label>
</fieldset><span class=\\"add-a-class\\" hidden=\\"\\"></span>
<fieldset class=\\"repeating_class-fieldset\\">
  <select name=\\"attr_select\\" title=\\"@{repeating_class-fieldset_$X_select}\\">
    <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
    <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
  </select>
  <label class=\\"input-label\\"><span class=\\"input-label__text\\" data-i18n=\\"select label\\"></span>
    <select class=\\"input-label__input\\" name=\\"attr_labelled_select\\" title=\\"@{repeating_class-fieldset_$X_labelled_select}\\">
      <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
      <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
    </select>
  </label>
</fieldset>
<button class=\\"repcontrol-button repcontrol-button--add\\" name=\\"act_add-custom-fieldset\\" type=\\"action\\" title=\\"%{add-custom-fieldset}\\">
</button>
<button class=\\"repcontrol-button repcontrol-button--edit\\" name=\\"act_edit-custom-fieldset\\" type=\\"action\\" title=\\"%{edit-custom-fieldset}\\">
</button>
<fieldset class=\\"repeating_custom-fieldset\\">
  <select name=\\"attr_select\\" title=\\"@{repeating_custom-fieldset_$X_select}\\">
    <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
    <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
  </select>
  <label class=\\"input-label\\"><span class=\\"input-label__text\\" data-i18n=\\"select label\\"></span>
    <select class=\\"input-label__input\\" name=\\"attr_labelled_select\\" title=\\"@{repeating_custom-fieldset_$X_labelled_select}\\">
      <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
      <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
    </select>
  </label>
</fieldset><span class=\\"inline-fieldset\\" hidden=\\"\\"></span>
<fieldset class=\\"repeating_inline-fieldset\\">
  <input class=\\"display-control\\" name=\\"attr_display_state\\" value=\\"short-display\\" hidden=\\"\\" type=\\"radio\\" title=\\"@{repeating_inline-fieldset_$X_display_state}\\"/>
  <div class=\\"inline-fieldset__summary display-target\\">
    <label class=\\"pointer\\">
      <input name=\\"attr_collapse\\" value=\\"1\\" hidden=\\"\\" type=\\"checkbox\\" title=\\"@{repeating_inline-fieldset_$X_collapse}\\"/><span class=\\"inline-fieldset__summary__text\\" name=\\"attr_name\\" title=\\"@{repeating_inline-fieldset_$X_name}\\"></span>
    </label>
  </div>
  <input class=\\"display-control\\" name=\\"attr_display_state\\" value=\\"display\\" checked=\\"\\" hidden=\\"\\" type=\\"radio\\" title=\\"@{repeating_inline-fieldset_$X_display_state}\\"/>
  <div class=\\"inline-fieldset__detail display-target\\">
    <input class=\\"collapse\\" name=\\"attr_collapse\\" value=\\"1\\" type=\\"checkbox\\" title=\\"@{repeating_inline-fieldset_$X_collapse}\\"/>
    <select name=\\"attr_select\\" title=\\"@{repeating_inline-fieldset_$X_select}\\">
      <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
      <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
    </select>
    <label class=\\"input-label\\"><span class=\\"input-label__text\\" data-i18n=\\"select label\\"></span>
      <select class=\\"input-label__input\\" name=\\"attr_labelled_select\\" title=\\"@{repeating_inline-fieldset_$X_labelled_select}\\">
        <option value=\\"option 1\\" data-i18n=\\"option 1\\" selected=\\"\\"></option>
        <option value=\\"option 2\\" data-i18n=\\"option 2\\"></option>
      </select>
    </label>
  </div>
</fieldset>
<button class=\\"repcontrol-button repcontrol-button--add repcontrol-button--inline\\" name=\\"act_add-inline-fieldset\\" type=\\"action\\" title=\\"%{add-inline-fieldset}\\">
</button>
<button class=\\"repcontrol-button repcontrol-button--edit repcontrol-button--inline\\" name=\\"act_edit-inline-fieldset\\" type=\\"action\\" title=\\"%{edit-inline-fieldset}\\">
</button>"
`;
